internal-dns/src/resolver.rs

```diff
@@ -201,27 +201,16 @@ impl Resolver {
         // SRV records have a target, which is itself another DNS name that
         // needs to be looked up in order to get to the actual IP addresses.
         // Many DNS servers return these IP addresses directly in the response
-        // to the SRV query as Additional records.  Ours does not.  See
-        // omicron#3434.  So we need to do another round of lookups separately.
+        // to the SRV query as Additional records.  Ours does as well.
+        // Unfortunately, trust-dns does not properly handle multiple targets.
```

Comment:
Is there a trust-dns issue for this?  (Is it still trust-dns#1980?)  I think it's good practice to reference the known issues here so that future readers know when this can be removed.

Reply:
Yea, it's the same FIXME they pointed out in that issue. I pulled out some of the code here to reuse in other places and moved the comment there as well. Hopefully it's a bit more clear now.